File: PlacementCellBackend/Controllers/RestaurentController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.Models;
using PlacementCellBackend.Services.Interfaces;

namespace PlacementCellBackend.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class RestaurantController : ControllerBase
    {
        private readonly IRestaurantService _restaurantService;

        public RestaurantController(IRestaurantService restaurantService)
        {
            _restaurantService = restaurantService;
        }

        [HttpGet]
        public async Task<ActionResult<IEnumerable<Restaurents>>> GetAllRestaurants()
        {
            var restaurants = await _restaurantService.GetAllRestaurantsAsync();
            return Ok(restaurants);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<Restaurents>> GetRestaurantById(int id)
        {
            var restaurant = await _restaurantService.GetRestaurantByIdAsync(id);
            if (restaurant == null)
                return NotFound();
            return Ok(restaurant);
        }

        [HttpPost]
        public async Task<ActionResult<Restaurents>> CreateRestaurant(Restaurents restaurant)
        {
            var created = await _restaurantService.CreateRestaurantAsync(restaurant);
            return CreatedAtAction(nameof(GetRestaurantById), new { id = created.restaurentid }, created);
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> UpdateRestaurant(int id, Restaurents restaurant)
        {
            if (id != restaurant.restaurentid)
                return BadRequest();

            var success = await _restaurantService.UpdateRestaurantAsync(id, restaurant);
            if (!success)
                return NotFound();

            return NoContent();
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteRestaurant(int id)
        {
            var success = await _restaurantService.DeleteRestaurantAsync(id);
            if (!success)
                return NotFound();

            return NoContent();
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/CRUD/RestaurentController.java
package com.example.placementicsbackend.controllers.CRUD;

import com.placementcellbackend.models.Restaurents;
import com.placementcellbackend.services.crud.interfaces.IRestaurantService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/restaurant")
@RequiredArgsConstructor
public class RestaurantController {

    private final IRestaurantService restaurantService;

    @GetMapping
    public ResponseEntity<List<Restaurents>> getAllRestaurants() {

        List<Restaurents> restaurants =
                restaurantService.getAllRestaurants();

        return ResponseEntity.ok(restaurants);
    }

    @GetMapping("/{id}")
    public ResponseEntity<Restaurents> getRestaurantById(
            @PathVariable Integer id) {

        Restaurents restaurant =
                restaurantService.getRestaurantById(id);

        if (restaurant == null) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.ok(restaurant);
    }

    @PostMapping
    public ResponseEntity<Restaurents> createRestaurant(
            @RequestBody Restaurents restaurant) {

        Restaurents created =
                restaurantService.createRestaurant(restaurant);

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(created);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Void> updateRestaurant(
            @PathVariable Integer id,
            @RequestBody Restaurents restaurant) {

        if (!id.equals(restaurant.getId())) {
            return ResponseEntity.badRequest().build();
        }

        boolean success =
                restaurantService.updateRestaurant(id, restaurant);

        if (!success) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.noContent().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteRestaurant(
            @PathVariable Integer id) {

        boolean success =
                restaurantService.deleteRestaurant(id);

        if (!success) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.noContent().build();
    }
}